OrbitBase/include/OrbitBase/ThreadUtils.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_THREAD_UTILS_H_
+#define ORBIT_BASE_THREAD_UTILS_H_
+
+#include <string>
+
+namespace orbit_base {
+
+#ifdef _WIN32
+using thread_id_t = uint32_t;
```

Comment:
This bothers me... we currently have a lot of windows code dealing with linux pid_t, so this is not really platform specific it is specific to the platform we are receiving information from. This might become an inconvenience where we are forced to do a lot of static casts or this type will become unused.

Reply:
@dpallotti , @dimitry- , again, thanks, please see the modifications.

I understand the concern about using a type alias, and I have made the change... I do find it annoying not to have the same type on Windows and Linux. I am tempted to change the Windows thread id type to int32_t and assert that we can do the conversion from uint32_t (in practice, the thread ids on Windows won't reach 2^31). A thread id in our code base would simply be int32_t and It would simplify the interface. Inversely, asserting that the thread id on Linux is always positive could work too... Anyhow, I've spent enough time on this PR for now.

Reply:
I mean, it surely is very annoying. But that's unfortunately one of those differences among the two platforms that are just there. Personally I'm more strongly against checking that the tid is positive on Linux, not because it isn't, but because the type is signed and we rely on that (for example, in some cases `perf_event_open` gives a `-1` pid), as most of the code that deals with thread ids is Linux as Linux is the target of profiling. I guess for the time being there wouldn't be a downside of assuming and checking that a tid on Windows fits in a signed int (unless of course it can happen to have tids above 2^31). It is still quite an assumption though, so I don't know.